src/main/java/lazyprogrammer/jwtdemo/security/jwt/JwtUtil.java

```diff
@@ -27,6 +27,7 @@ private SecretKey getSigningKey() {
 
     public String generateToken(UserDetails userDetails) {
         Map<String, Object> claims = new HashMap<>();
+        claims.put("ROLES", userDetails.getAuthorities());
```

Comment:
Because we are fetching the roles from UserDetails while setting Authorities in Principal, this claim will not be used.